core/products/future_test.go

```diff
@@ -36,47 +36,125 @@ func TestScalingOfSettlementData(t *testing.T) {
 }
 
 func testNoScalingNeeded(t *testing.T) {
-	ft := testFuture(t)
+	// Create test future with settlement data type integer with decimals (that represents a decimal)
+	ft := testFuture(t, datapb.PropertyKey_TYPE_INTEGER)
 
 	// settlement data is in 5 decimal places, asset in 5 decimal places => no scaling
-	scaled, err := ft.future.ScaleSettlementDataToDecimalPlaces(num.NewUint(100000), 5)
+	scaled, err := ft.future.ScaleSettlementDataToDecimalPlaces(
+		&num.Numeric{
+			AsUint: num.NewUint(100000),
+		}, 5,
+	)
 	require.NoError(t, err)
 	require.Equal(t, num.NewUint(100000), scaled)
+
+	// Create test future with settlement data type decimal with decimals (that represents a decimal)
+	ft = testFuture(t, datapb.PropertyKey_TYPE_DECIMAL)
+
+	// settlement data is in 5 decimal places, asset in 3 decimal places => x10^-2
+	dec := num.DecimalFromFloat(10000.01101)
+	scaled, err = ft.future.ScaleSettlementDataToDecimalPlaces(
+		&num.Numeric{
+			AsDecimal: &dec,
+		}, 5,
+	)
+	require.NoError(t, err)
+	require.Equal(t, num.NewUint(1000001101), scaled)
 }
 
 func testScalingUpNeeded(t *testing.T) {
-	ft := testFuture(t)
+	// Create test future with settlement data type integer with decimals (that represents a decimal)
+	ft := testFuture(t, datapb.PropertyKey_TYPE_INTEGER)
 
 	// settlement data is in 5 decimal places, asset in 10 decimal places => x10^5
-	scaled, err := ft.future.ScaleSettlementDataToDecimalPlaces(num.NewUint(100000), 10)
+	scaled, err := ft.future.ScaleSettlementDataToDecimalPlaces(
+		&num.Numeric{
+			AsUint: num.NewUint(100000),
+		}, 10,
+	)
 	require.NoError(t, err)
 	require.Equal(t, num.NewUint(10000000000), scaled)
+
+	// Create test future with settlement data type decimal with decimals (that represents a decimal)
+	ft = testFuture(t, datapb.PropertyKey_TYPE_DECIMAL)
+
+	// settlement data is in 5 decimal places, asset in 3 decimal places => x10^-2
+	dec := num.DecimalFromFloat(10000.00001)
+	scaled, err = ft.future.ScaleSettlementDataToDecimalPlaces(
+		&num.Numeric{
+			AsDecimal: &dec,
+		}, 10,
+	)
+	require.NoError(t, err)
+	require.Equal(t, num.NewUint(100000000100000), scaled)
 }
 
 func testScalingDownNeeded(t *testing.T) {
-	ft := testFuture(t)
+	// Create test future with settlement data type integer with decimals (that represents a decimal)
+	ft := testFuture(t, datapb.PropertyKey_TYPE_INTEGER)
 
 	// settlement data is in 5 decimal places, asset in 3 decimal places => x10^-2
-	scaled, err := ft.future.ScaleSettlementDataToDecimalPlaces(num.NewUint(100000), 3)
+	scaled, err := ft.future.ScaleSettlementDataToDecimalPlaces(
+		&num.Numeric{
+			AsUint: num.NewUint(100000),
+		}, 3,
+	)
 	require.NoError(t, err)
 	require.Equal(t, num.NewUint(1000), scaled)
+
+	// Create test future with settlement data type decimal with decimals (that represents a decimal)
+	ft = testFuture(t, datapb.PropertyKey_TYPE_DECIMAL)
+
+	// settlement data is in 5 decimal places, asset in 3 decimal places => x10^-2
```

Comment:
This test makes no sense. If the settlement data is already in decimal format, then we should not have a specific number of decimal places. We should take whatever decimal value from the external source, and scale it to the uint in asset precision. In this example, with a value of `10000.00001` (5 decimal places) should yield `1000000001` (if the asset has 5 decimal places), or `100000000100000` (for an asset with 10 decimal places) and so on

Reply:
Not sure what do you mean.
The decimal places that are sent via the parameter here are the asset decimals.
When the settlement data type is decimal - the place set inside `NumberOfDecimalPlaces` field are not taken into the calculation -> https://github.com/vegaprotocol/vega/pull/6935/commits/db7f4cd7157e7edeeb1c9c464b9094521d5be6fb#diff-01e84bbc9c49dbdd4b9e45ec724423ea40a7c808577eebb58ed4f69f25ac4755R68-R75
But only the target decimals - in this case 3.

And when the settlement data is updated with a decimal type - they are also not set in the new object -> https://github.com/vegaprotocol/vega/pull/6935/files#diff-72277c761cbd8dabe6ed10e8f5128682486134c673a42959dca45dcff971df3fR221
and upon creation are also not taken -> https://github.com/vegaprotocol/vega/pull/6935/files#diff-72277c761cbd8dabe6ed10e8f5128682486134c673a42959dca45dcff971df3fR269-R271
Or you meant that this test is artificial? But I dont know why it would be

Reply:
`10000.00001` being the settlement data for an asset with only 3 decimal places doesn't make sense. If the settlement data is a value in asset X, it can only be valid if it has no more than 3 decimals.